docs/user-documentation/soc-and-feeds.md

```diff
@@ -0,0 +1,126 @@
+
+```ts
+type SyncSigner = (digest: Uint8Array) => Signature
+type AsyncSigner = (digest: Uint8Array) => Promise<Signature>
+
+/**
+ * Interface for implementing Ethereum compatible signing.
+ *
+ * @property sign     The sign function that can be sync or async
+ * @property address  The ethereum address of the signer
+ */
+export type Signer = {
+  sign: SyncSigner | AsyncSigner
+  address: EthAddress
+}
+```
+
+:::warning Your communication privacy may be at risk
+We suggest to use either ephemeral private keys (e.g. randomly generated) or the `Signer` interface when writing to SOC or Feeds. Never use your real Ethereum private keys here (or in any web applications really) directly because you may lose all your Ethers belonging to it.
+:::
+
+Using the writer interface is then similar to using the reader:
+
+```js
+const signer = '0x634fb5a872396d9693e5c9f9d7233cfa93f395c093371017ff44aa9ae6564cdd'
+const socWriter = bee.makeSOCWriter(signer)
+const identity = '0000000000000000000000000000000000000000000000000000000000000000'
+const data = new Uint8Array([1, 2, 3])
+const response = await socWriter.upload(identity, data)
+```
+
+## Feeds
+
+Feeds are an abstraction built on top of SOCs to provide mutable resources on the otherwise immutable data types that Swarm supports.
+
+A feed is defined by its `owner` (see above), a `topic` (which is 32 bytes arbitrary data, usually expected as a hex string or in a `Uint8Array` type) and a `type`. `type` defines how the index of the feed is stored in Swarm and how the lookup works, currently only the `sequence` type is supported.
+
+Publishers are the single owners of feed chunks and are the only ones able to post updates to their feed. Posting an update requires (1) constructing the identifier from the topic and the correct index and (2) signing it concatenated together with the hash of the arbitrary content of the update.
+
+Conversely, users can consume a feed by retrieving the chunk by its address. Retrieving an update requires the consumer to construct the address from the owner’s public key and the identifier. To calculate the identifier they need the topic and the appropriate index. For this they need to know the indexing scheme.
+
+Feeds enable Swarm users to represent a sequence of content updates. The content of the update is the payload that the feed owner signs against the identifier. The payload can be a swarm reference from which the user can retrieve the associated data.
+
+### Topic
+
+In Swarm `topic` is a 32-byte arbitrary byte array. It's possible to choose an arbitrary topic for each application and then knowing someone's (or something's) address it's possible to find their feeds. Also this can be the hash of one or more human readable strings, specifying the topic and optionally the subtopic of the feed. There is a helper function provided for that:
```

Comment:
```suggestion
In Swarm `topic` is a 32-byte long arbitrary byte array. It's possible to choose an arbitrary topic for each application and then knowing someone's (or something's) address it's possible to find their feeds. Also this can be the hash of one or more human readable strings, specifying the topic and optionally the subtopic of the feed. There is a helper function provided for that:
```